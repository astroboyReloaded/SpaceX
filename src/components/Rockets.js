import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rocketsSlice';

const Rockets = () => {
  const dispatch = useDispatch();
  const { rockets, loading, error } = useSelector((state) => state.rockets);

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  return (
    <>
      <h1>Rockets</h1>
      <section>
        {loading && <span>loading...</span>}
        {rockets.length && rockets.map((rocket) => (
          <article key={rocket.id}>{rocket.name}</article>
        ))}
        {error && <span>error</span>}
      </section>
    </>
  );
};

export default Rockets;
