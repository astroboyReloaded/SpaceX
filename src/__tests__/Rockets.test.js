import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/Store';
import Rockets from '../components/rockets/Rockets';

it('Renders Rockets', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={<Rockets />} />
          </Routes>
        </Router>
      </Provider>,
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
