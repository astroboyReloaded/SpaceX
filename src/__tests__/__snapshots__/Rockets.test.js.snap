// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Rockets 1`] = `<section />`;
